src/main/java/com/nextstep/lotto/calculator/StringAddCalculator.java

```diff
@@ -0,0 +1,56 @@
+package com.nextstep.lotto.calculator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final int ZERO = 0;
+    private static final String CUSTOM_REGEX = "//(.)\n(.*)";
+
+    public static int splitAndSum(String source) {
+        if (source == null || "".equals(source)) {
+            return ZERO;
+        }
+
+        CalculatorItem item = parse(source);
+        return addItem(item);
+    }
+
+    private static CalculatorItem parse(String source) {
+        if(!source.matches(CUSTOM_REGEX)) {
+            return new CalculatorItem(source);
+        }
+        return parseCalculatorItem(source);
+    }
+
+    private static CalculatorItem parseCalculatorItem(String source) {
+        Pattern pattern = Pattern.compile(CUSTOM_REGEX);
+        Matcher matcher = pattern.matcher(source);
+        CalculatorItem item = null;
+        if (matcher.find()) {
+            String delimiter = matcher.group(1);
+            String numbers = matcher.group(2);
+            item = new CalculatorItem(numbers, delimiter);
+        }
+        return item;
```

Comment:
현재 로직상으로는 null을 반환할 일은 없지만 새로운 기능 추가나 수정사항을 반영하다보면 문제가 될 수 있습니다
null 자체를 반환할 여지를 안만들면 NPE가 발생할 일이 없지 않을까요?

```suggestion
        if (matcher.find()) {
            String delimiter = matcher.group(1);
            String numbers = matcher.group(2);
            return new CalculatorItem(numbers, delimiter);
        }
        return new CalculatorItem(source);
```

Reply:
네 의견 감사합니다 😄